gateway-discovery-cm/src/main/java/org/apache/knox/gateway/topology/discovery/cm/DiscoveryApiClient.java

```diff
@@ -157,12 +161,26 @@ private String getUsername() {
     return username;
   }
 
-  private void configureTruststore(KeystoreService keystoreService) {
-    SSLContext truststoreSSLContext = TruststoreSSLContextUtils.getTruststoreSSLContext(keystoreService);
+  private void configureSsl(GatewayConfig gatewayConfig, KeyStore trustStore) {
+    final SSLContext truststoreSSLContext = TruststoreSSLContextUtils.getTruststoreSSLContext(trustStore);
+
     if (truststoreSSLContext != null) {
+      final ConnectionSpec.Builder connectionSpecBuilder = new ConnectionSpec.Builder(ConnectionSpec.MODERN_TLS);
+      if (gatewayConfig.getIncludedSSLCiphers().isEmpty()) {
```

Comment:
Can this method `getIncludedSSLCiphers()` return null if no `ssl.include.ciphers` is configured?

Reply:
Nope; it cannot be null (either an empty collection or one with the declared items).

Reply:
I might miss something but I don't see how is this enforced. What if there is no `SSL_INCLUDE_CIPHERS` in the config?

```
  @Override
  public List<String> getIncludedSSLCiphers() {
    List<String> list = null;
    String value = get(SSL_INCLUDE_CIPHERS);
    if (value != null && !value.isEmpty() && !"none".equalsIgnoreCase(value.trim())) {
      list = Arrays.asList(value.trim().split("\\s*,\\s*"));
    }
    return list;
  }
```

Reply:
Ohh...and you are right. I added a new method - `getIncludedSSLProtocols` - I misread and thought your question was about that one. Let me submit a new PS soon.

Reply:
Fixed.